test/mocks/upstream/cluster_info.h

```diff
@@ -147,6 +147,7 @@ class MockClusterInfo : public ClusterInfo {
   MOCK_METHOD(bool, drainConnectionsOnHostRemoval, (), (const));
   MOCK_METHOD(bool, connectionPoolPerDownstreamConnection, (), (const));
   MOCK_METHOD(bool, warmHosts, (), (const));
+  MOCK_METHOD(bool, setLocalInterfaceNameOnUpstreamConnections, (), (const));
```

Comment:
This looks almost identical to the original PR, so I can rubber stamp the backport. But, the tests in protocol_integration_test aren't here. Can you comment on these and any other delta? Thanks.

Reply:
Not sure how I missed that. One of the test points actually fails in my workspace. Investigating it .

Reply:
Any update on this @pradeepcrao ?